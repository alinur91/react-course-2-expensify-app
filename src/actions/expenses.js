import uuid from 'uuid';

// ADD_EXPENSE
export const addExpense = (
    {
        description='',note='', /* default values */
        amount=0,
        createdAt=0
    }={}
) => ({ 
/* if object doesnt exist we will desctrutre empty object */
    type: 'ADD_EXPENSE',
    expense: {
        id: uuid(),
        description,
        note,
        amount,
        createdAt
    }
})

//REMOVE_EXPENSE
export const removeExpense = ({id}={}) =>( /* {id}={} degen zna4it dispatch(removeExpense({id})) degen soz object peredaem */
    {
        type: 'REMOVE_EXPENSE',
        id
    }
)

//EDIT_EXPENSE
export const editExpense = (id,updates) =>(
    {
        type: 'EDIT_EXPENSE',
        id,
        updates
    }
)