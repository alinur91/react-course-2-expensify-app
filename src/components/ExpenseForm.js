import React from 'react';
import moment from 'moment';
import {SingleDatePicker} from 'react-dates'; /* SingleDatePicker degen component */
/* SingleDatePicker component do have required props */
//const date = new Date()
// const now = moment() /*we get back instance of moment,if u don't provide anything its now  */
// console.log(now.format('MMM Do, YYYY')) /* format() method its going to print formatted version of the now */


export default class ExpenseForm extends React.Component {
/*The goal is to take these 4 state values and only use  defaults if no expense passed down  */
/* if an expense was passed down we want to start these off that those values(props.expense) */
/*To get that done we have to look at props.We have to define our state in the constructor function to access the props  */
//props.expense.createdAt timeStamp.<ExpenseForm/> componenty we reuse v AddExpensePage i EditExpensePage componentax poetomy kogda v <AddExpensePage/> componente use descrip,amount bari empty boladi,kogda v <EditExpensePage/>-e ispolzuem ExpenseForm dannyie dolzny byt populated
constructor(props){ /* state vklu4ili v constr function shtoby byl access to props() */
    super(props)
    this.state = {
        description: props.expense?props.expense.description:'',
        note: props.expense?props.expense.note:'',
        amount: props.expense?(props.expense.amount / 100).toString():'',
        createdAt: props.expense?moment(props.expense.createdAt):moment(),
        calenderFocused: false,
        error: ''
    }
}
    
    onDescriptionChange=e =>{
  
        const description= e.target.value /* inputtin valuei descrip: 'asd' */
        this.setState(()=>({description})) /*img gonna pass in updated function,will return implicitly object  */
    }
    onNoteChange = e=>{ /* kazhdyi raz kak e.target.value changes v object {note: e.target.value} budet i textarea nadpis budet 4itat s value={this.state.note} statea */
        const note =e.target.value
        // e.persist()
        this.setState(()=>({note}))
    }
    onAmountChange = e =>{
        /* we're not going to blindly set,conditional logic bolady */
        const amount = e.target.value; /*e.target.value; degen svezhyi value nado setStatepen v objecte zapisat  */
//!amount degen empty string bolsa(falsy) vse ravno input value set it isteimiz
        if(!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)){this.setState(()=>({amount}))}
        /* tobedegi crietria shak kelmese we're not going to setState */
    }/* amount.match(/^\d*(\.\d{0,2})?$/) degen regular expression 12.12 boly kerek nukteden keyin 2 cifra kerek */
/* the big picture goal here is to use local component state to track the changes to all of these inputs,only when the user submits the form will do smth with that information */
/* When they submit the from will send that off to redux to either edit the existing expense or create the new one*/
/* value={this.state.description}} degen creates a read only input,we want to user to be able to change the value */
onDateChange = createdAt =>{ /* kogda user menyaet daty,on menyaetsya na to shto on postavil,createdAt degen shto on postavil */
/*onDateChange function boladi,its got to get called with moment() instance when someone picks a new day from the calendar  */
if(createdAt){this.setState(()=>({createdAt}))}
    /* if there is no createdAt(esli udalit delete keymen) do not nothing */
/* this is going to prevent the user from being able to clear that value */
}
onFocusChange= ({focused}) =>{
    
    this.setState(()=>({calenderFocused:focused}))
}
onSubmit = e =>{
    e.preventDefault() /* full page refesh bolmaidy kogda button baskan kezi */
/* description men amount required field,calendarmen,note optional */
    if(!this.state.description || !this.state.amount){ /* on uzhe napisal description i amount field,kogda submit nazhimaet proveryaem esli ili net descr,amount,esli net 1-o iz dvux */
/* descr ili amount zhazbasa,hot' 1-ein,we want to render error to the screen */
        // Set error state equal to ''
        this.setState(()=>({error: 'Please provide description and amount.'}))

    }else{ /* if there is no error */
        // esli 1-yi raz s errorom bil mi hotim za4istit error
         this.setState(()=>({error: ''})) 
         this.props.onSubmit({ /* zdes peredaem object,v AddExpensePage prinimaem object dispatchim shtoby renderilsya expense,i redirecr props.history.push('/') */
             description: this.state.description,
             amount: parseFloat(this.state.amount) * 100, /*this.state.note degen 12.12 nado preobrazavat  */
             note: this.state.note,
             createdAt: this.state.createdAt.valueOf(), /*this.state.createdAt degen moment object  */
            
         }) /*this.props degen 4erez instance props onSubmit peredali <ExpenseForm  onSumbit(expense=>{console.log(expense)})/>onSubmit degen callback koi 4erep props peredaem i sdes vizivaem a v propse v functione polu4im expense,shtoby store.dispatch(addExpense(expense)) sdelat 4toby za renderit etot expense  */
    }
}
    render() { /*let's setup onSubmit handler for the form,kogda submit sdelaet redirect boladi dashboard page-ge  */
        return ( /* kogda addExpense nazhimaet button vizivaetsya onSubmit function i peredaet object a v EditExpesePage-e prinimaem onSubmit(expense)=>{} */
            <div>
                {this.state.error && <p>{this.state.error}</p>}
                <form onSubmit={this.onSubmit}>
                    <input 
                        value={this.state.description} 
                        autoFocus 
                        placeholder="Description"
                        onChange={this.onDescriptionChange}
                    /> {/* autoFocus is going to make sure that when we visit the page it automatically puts the cursor and focus right in input */}
                    <input 
                        type="text" 
                        placeholder="Amount"
                        value={this.state.amount}
                        onChange={this.onAmountChange}
                    />
                    <SingleDatePicker
                        date={this.state.createdAt}
                        onDateChange={this.onDateChange}
                        focused={this.state.calenderFocused}
                        onFocusChange={this.onFocusChange}
                        numberOfMonths={1}
                        isOutsideRange={(day)=> false} 
                     />
                    <textarea 
                        value={this.state.note}  
                        onChange={this.onNoteChange}
                        placeholder="Add a note for your expense (optional)">
                    </textarea>
                    <button>Add Expense</button>
                </form>
            </div>
        )
    }/*date degen-moment object that represents where you want to start,we wanna start at now,when the user loads the page,bugingi kundi korsetemiz  */
//numberOfMonths={1} degen calendarda 1 ai korsetedi.isOutsideRange={(day)=> false}  degen otken shakti calendarda korsetedi
/* onFocusChange degen handler when react dates library needs to change that value */
}


/*How to simulate user interaction in your test cases.If I submit the form with bad data I would expect that error gets set and rendered to the screen. */
//We will write test case that makes sure an error gets rendered when the form is submitted without a description/an amount