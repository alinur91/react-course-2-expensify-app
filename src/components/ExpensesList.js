/* statleess functional component */
import React from 'react';
import {connect} from 'react-redux'; /*its called connect because it connects your component to the redux store  */
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses'

/* We do have the ability to use connect to pull data off but we haven't done that yet */
/* named export isteimyz shtoby protestit */
export const ExpensesList = props =>( /*props.expenses degen=state.expenses,state.expenses degen store.expenses,const createStore(combineReducers({expenses: expensesReducer, filters: filtersReducer}))  */
/* We've connected our 1st react component to the redux store */
/*expense degen {description:'asd',amount:123,createdAt: 44} ego mozhno object destructure pryam v instance ExpenseListItem polu4itstya <ExpenseListItem key={expense.id} description={expense.description},amount:{expense.amount},createdAt: {expense.createdAt} />   */
/*props.expenses degen---> expenses: selectExpenses(state.expenses,state.filters),minay degen selectExpenses(state.expenses,state.filters) ==> [{desc:'a',amount:123}]  */
// {...expense}/>) /*{...expenses} degen ===> <ExpenseListItem id='123' description='rent' amount=900 />  */
    <div> 
        {
            props.expenses.length === 0 ? (
                <p>No expenses</p>
            ): (
                props.expenses.map(expense=>(<ExpenseListItem key={expense.id} {...expense}/>))
            )
        } 
    </div>
)/*  dispatch(removeExpense({id})) delaem shtoby combineStore(expenses: ExpenseReducer) zarobatal vernet updated array, i s pomoshu connecta berem stae i renderem v componente ExpensesList renderim */
/* We're going to be using the connect function and all of our individual components */
/* Components like ExpensesList that need to either dispatch actions or read from the store */

/*Lets focus reading from the store.We have 2 items that are getting added to the list.Water bill,Gas bill  */
//Try to get those 2 items showing up
// Step one-->Create a new const for the higher order component.We get smth back from connect(),we do not get back higher order component
// we need to call that funct with the component,Why even have this argument at all.This is where we provide the information about what we want to connect
/* So we have a ton of information on the store does the component need all of it.No! it usually just needs a subset. */

const mapStateToProps =state=>{ /* state degen {expenses: [], filters:{}}, v app.js my sdelali store.dispatch(addExpense({description: 'Water bill',amount: 100,createdAt: 4444})) i store.dispatch(addExpense({description: 'Gas bill',createdAt: 444})) i store.dispatch(setTextFilter('bil'))   */
/*This function lets us determine what information from the store we want our component to be able to access  */
/* and stores state gets passed in  */ /* reading off of state */
return { /* now ConnectedExpenseList component is going to have access to the name prop */
    expenses: selectExpenses(state.expenses,state.filters) /*selectExpenses vizivaem potomy shto, store.dispatch(setTextFiler('bill')) sdelali, i sdez selectExpenses vernet etot updated array im going to passed that down to ExpenseList,selectExpenses vernet filtered,sorted array [{description:'a',id:12,amount:'bb'}] */
    
}
} /* So as the store changes this is automatically going to rerun,getting the fresh values in the component */

//When you connect a component to the redux store its reactive which means as the store changes your component is going to get re rendered with those new values
// This <ExpensesList/> doesnt need to worry about using store.subscribe or store.getState()
//Doesnt have to use any components state to manage that data.Instead all of that is done for us by react-redux

/* Mi zaconnectili <ExpensesList/> component k store */
export default  connect(mapStateToProps) (ExpensesList) // /*ExpensesList componente budet dostyp k expenses [{}] 4erez props.expenses  */
/* esli v store proishodit izmenenie,za connectinny componenta <ExpensesList/> zapuskaet connect() i vnutri <ExpensesList/> re-renderitsya obnovlennie stori  */
/* sdez delaem const ExpensesListConnected export shtoby use v ExpenseDashboardPage.js <ExpensesList /> */

/* We're going to learn how he can dispatch from a connected componenta <ExpensesList/>  */