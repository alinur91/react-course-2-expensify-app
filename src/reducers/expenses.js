// Expenses Reducer
const expensesReducerDefaultState = []

export default  (state=expensesReducerDefaultState ,action) =>{
    // console.log(state)
/* return a new state? */
switch (action.type) { /* push changes original array,we don't want to change state or action,we just want to read off of it */
case 'ADD_EXPENSE': return [...state,action.expense]//state degen: [{id:123,description:'asd',note:'',amount:123,createdAt: 456}] action.expense degen {id:123,description:'asd',note:'',amount:123,createdAt: 456} /* returns a new array,return array soxranitsya k expensesReducer variablu */
/* aside from concat we can use spread operator.action degen object vnutri property expense,id,amount */
    case 'REMOVE_EXPENSE': return state.filter(({id})=> id !==action.id)
    case 'EDIT_EXPENSE': return state.map(expense=>{
/* action degen   {type: 'EDIT_EXPENSE',id,updates} updates degen {amount: 900} */
/* state degen [{id: "bed68040-5eed-4835-81d9-1ec45e9f06bf", description: "Rent", note: "rent for may", amount: 800, createdAt},{id: "9f0a4e1c-1f5c-4085-a425-277adb9116e8", description: "Buy a car", note: "", amount: 10000, createdAt: 0}] */
        if(expense.id === action.id){
            return { /* i want to return a new object,changing expense,i want to grab all existing properties,and overwrite any of the ones that were passed down */
                ...expense,
                ...action.updates
            }
        }else{
            return expense
        }
    })
    default: return state    
}
}

