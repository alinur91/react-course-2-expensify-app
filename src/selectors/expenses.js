import moment from 'moment';

// Get visible expenses
export default (expenses,{text,sortBy,startDate,endDate}) => { /* vse dispatch sdelali neskolko zdes polnaya informacia,s pomoshui etogo vernem array [{},{}] */
    return expenses.filter(expense=>{
const createdAtMoment = moment(expense.createdAt) //expense.createdAt cifra goi
const startDateMatch= startDate? startDate.isSameOrBefore(createdAtMoment,'day') : true; //if there is no start date we're never going to filter things out based off of that,startDate always be true if there is no startdate
//typeof startDate !== 'number' || expense.createdAt >= startDate; /*only a start date is a number do we want to filter expenses out  */
/*lets say startdate is undefined,and createdAt is 1,it will stay(wont be filtered).Lets say startDate is 2 and created_at is 1(it will not stay,sozdannyi data ranshe bil,a user pozhe pokazivaet startDate) etot false budet startDateMatch  */
const endDateMatch= endDate? endDate.isSameOrAfter(createdAtMoment,'day') : true;
//typeof endDate !== 'number' || expense.createdAt <= endDate;
const textMatch=expense.description.toLowerCase().includes(text.toLowerCase())

        
/* esli vse eto startDateMatch && endDateMatch && textMatch bydet true to  expense ostaetsya v arraye*/
        return startDateMatch && endDateMatch && textMatch
    }).sort((a,b)=>{ /* a,b degen object {id: 123,description:'asd',createdAt: 123,amount:123} */
        if(sortBy === 'date') { /* sortiruem po recent data */
            return a.createdAt > b.createdAt?-1:1
        } // I want to see my most expenses coming 1st,from sorting date i want to see recent expenses coming 1st
        else if(sortBy === 'amount'){
            return b.amount - a.amount /* [1(a),2(b)] b-a = polozhitelnoe chislo,zna4it menyaem mestami,sonda kemy retimen boladi 800,100,50 */
           // a.amount > b.amount?-1:1
        }
    })
}

//JEST testing framework