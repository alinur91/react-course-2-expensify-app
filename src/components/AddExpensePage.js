import React from 'react';
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {addExpense} from '../actions/expenses'

export class AddExpensePage extends React.Component{
  onSubmit=expense=>{ /* it takes the data and passed it through */
    // props.dispatch(addExpense(expense))
     this.props.addExpense(expense)
     this.props.history.push('/') /*Components we render inside of react-router they get a bunch of special props==> props.history.push  */
{/* dashboard pagege redirect / ,this is going to switch me over as if i clicked the link,which means we're not going through full page refresh  */}
   }
   render(){
     return (
        /* onSubmit ishindegi function gets called when the form gets submitted,with valid data
 and we get that data back,expense {description:'asd',amount:'asd'},4ere props peredaem onSumbut*/
 /*this.props degen 4erez instance props onSubmit peredali <ExpenseForm  onSumbit(expense=>{console.log(expense)})/>onSubmit degen callback koi 4erep props peredaem i sdes vizivaem a v propse v functione polu4im expense,shtoby store.dispatch(addExpense(expense)) sdelat 4toby za renderit etot expense  */
 /* expense page needs to be able to dispatch the given action to the redux store */
    <div> 
      <h1>Add Expense</h1>
        <ExpenseForm onSubmit={this.onSubmit}/> {/*push degen kak-budto link basty birey  */}
    </div>
     )
   }
}

const mapDispatchToProps = (dispatch) =>({
    addExpense: (expense)=> dispatch(addExpense(expense))
  })
//the goal is to return an object
  /* on here we define various props and these props are going to call dispatch */
    
/*When we're adding and edditing we can dispatch different stuff  */

/* 1st function is mapsStatesToProp=mynda undefined */
export default connect(undefined,mapDispatchToProps) (AddExpensePage);  
/* za connectili AddExpensePage componenty, and now we have access to props.dispatch*/
/* we do need to pass that component into the 2nd one */