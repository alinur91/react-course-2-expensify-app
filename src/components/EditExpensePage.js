import React from 'react';
import {connect} from 'react-redux'; /* hotim podklu4tisya k redux store
step2 --> it's going to be to search the expenses array for expense with and id that matches this one {props.match.params.id} */
import {editExpense,removeExpense} from '../actions/expenses';
import ExpenseForm from './ExpenseForm';
 
export class EditExpensePage extends React.Component {
  onSubmit= (expense) =>{ /* v <ExpenseForme/> vizivaem onSubmit() a zdes prinimaem */
//     this.props.onSubmit({ /* zdes peredaem object,v AddExpensePage prinimaem object dispatchim shtoby renderilsya expense,i redirecr props.history.push('/') */
//     description: this.state.description,
//     amount: parseFloat(this.state.amount) * 100, /*this.state.note degen 12.12 nado preobrazavat  */
//     note: this.state.note,
//     createdAt: this.state.createdAt.valueOf(), /*this.state.createdAt degen moment object  */
   
// }) /*this.props.expense.id degen minay goi ===> expense: state.expenses.find(expense=> expense.id === props.match.params.id)  */
    this.props.editExpense(this.props.expense.id,expense) /* zdez prosto vizivaem editExpense(expense),v mapDispatchToProps-e dispatchom peredaem*/
    this.props.history.push('/') /*Components we render inside of react-router they get a bunch of special props==> props.history.push  */
  }
  onRemove= (expense) =>{ //this.props.expense.id degen mapStateToProps functionda retun bolady {expense: {}}
    this.props.removeExpense({id: this.props.expense.id})
    this.props.history.push('/') /*Components we render inside of react-router they get a bunch of special props==> props.history.push  */
  }
  
  render(){
    console.log(this.props.expense)
    return ( /* onSubmit= {this.onSubmit} degen v <ExpenseForm/>e vizivaem onSubmit() */
      <div>
        <ExpenseForm
          expense ={this.props.expense}
         onSubmit= {this.onSubmit}
         />
         <button onClick={this.onRemove} >remove</button>
      </div>
    )
  }
}

/*The difference between the AddExpensePage and EditExpensePage is <AddExpensePage/> is fine with those empty defaults and EditExpensePage isn't fine */
/* It wants to be able to overwrite the current values for that expense  */
 /* HIGher order component pen EditExpensePage componenta both have access to props */
    // console.log(props) /* <Router path="/create" component={AddExpensePage}> passing stuff in and that stuff can be useful as we start building more real world app */
/*Any time we use a component inside of a route <BrowserRouter>  we get access to special information(like location,params,querystring)  */
/* kogda zahodim v etot put' <Route path="/edit/:id" component={EditExpensePage} /> my izvlekaem id with the help of {props.match.params.id} */
    
/*  props.history.push('/')  degen redirect to main page */

const mapStateToProps = (state,props) =>{ /* state degen [{descr:'water',amount:23},{}] */
/* props degen <Route path="/edit/:id" component={EditExpensePage} /> osy goi path */
/* what are we searching for? {props.match.params.id} osy id-di expense izdeb zhurmiz */
/*How do we access it? We have access to the props right here.EditExpensePage-tin propsi */
/*We can take some of the current props that were passed into the higher order component and we can use them to calculate the props that we want to add on  */
/* react router it renders our higher order component,higher order component passes the props through and it also allows us to add on some new ones  */
  return { /* return true bolsa callback ishinde 1-yi item tabady,false bolsa eshkaisysi */
    expense: state.expenses.find(expense=> expense.id === props.match.params.id)
  }/* tobedegi <EditExpensePage/> componenta 4erez props access boladi props.expense dep */
}

const mapDispatchToProps = (dispatch) =>(
  { /* we get the data in and pass it through expense v samomo componente vizvali this.props.editExpense(expense) */
    editExpense: (id,expense)=> dispatch(editExpense(id,expense)), //editExpense vnutri etogo functiona dispatchim
    removeExpense: (data)=> dispatch(removeExpense(data))
  }
)

/* nado za connectit etot <EditExpensePage/> componenty k redux store.
Potomy shto we want to be able to get the entire expense object*/
  export default connect(mapStateToProps,mapDispatchToProps) (EditExpensePage);