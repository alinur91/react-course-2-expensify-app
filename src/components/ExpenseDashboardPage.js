import React from 'react';
import ExpensesList from './ExpensesList' /*ExpensesList degen export default  connect(mapStateToProps) (ExpensesList)  */
import ExpenseListFilters from './ExpenseListFilters'

const ExpenseDashboardPage = () =>( /* <ExpensesList /> vernet <h1>Expense List</h1> <h3>Water bill.</h3> <p>123 - 455</p>*/ 
/*<ExpenseListFilters i <ExpensesList /> componety budut imet dostyp k redux store,4erez connect/>  */
    <div> 
      <ExpenseListFilters />
      <ExpensesList />
    </div>
  );
  
export default ExpenseDashboardPage;