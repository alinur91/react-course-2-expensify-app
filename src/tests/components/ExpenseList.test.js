import React from 'react'; /* Component(JSX) koldanamyz poeteomy import */
import {ExpensesList} from '../../components/ExpensesList';  /*Pass in props and make sure it renders correctly. we want test <ExpensesList /> unconnected version because we want to be able to provide a set of dynamic props */
import {shallow} from 'enzyme'; // import because we're going to be using that to render our component
import expenses from '../fixtures/expenses'; /* array vidumanniy koldanamyz */
//These are the imports that we need to snapshot test <ExpenseList/>
/* we dont want these props to come from the store,we're going to provide expenses directly */

test('should render ExpenseList with expenses',()=>{
//We're going to see what happens when we render expense list with given expenses(ush item v arraye vidaumannie)
    const wrapper = shallow(<ExpensesList expenses={expenses}/>)
    expect(wrapper).toMatchSnapshot() /* 1st time run there is no existing snapshot,test wil pass and will create a new snapshot,1 snapshot was added,esli ol snapshot will change we'll get notified */
})

//We could also create a test case to see what happens when we have no expenses,when we have that empty array
/*When there are expenses they show up and when there are no expenses we get our error message */
/*If either of those things change the snapshot test will fail,alerting us and we'll be able to respond accordingly  */
test('should render ExpenseList with empty message',()=>{
    const wrapper = shallow(<ExpensesList expenses={[]} />) /* what happens when we pass empty array? */
    expect(wrapper).toMatchSnapshot() /* 1st time run there is no existing snapshot,test wil pass and will create a new snapshot,1 snapshot was added,esli ol snapshot will change we'll get notified */

})