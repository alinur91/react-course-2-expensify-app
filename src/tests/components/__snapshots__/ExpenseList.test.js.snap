// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with empty message 1`] = `
<div>
  <p>
    No expenses
  </p>
</div>
`;

exports[`should render ExpenseList with expenses 1`] = `
<div>
  <ExpenseListItem
    amount={1120}
    createdAt={0}
    description="xbox"
    id="123"
    note=""
  />
  <ExpenseListItem
    amount={1950000}
    createdAt={-345600000}
    description="toyota camry 60 2020"
    id="456"
    note=""
  />
  <ExpenseListItem
    amount={4500}
    createdAt={777600000}
    description="motocycle"
    id="789"
    note=""
  />
</div>
`;
